manager/api/transport_test.go

```diff
@@ -0,0 +1,535 @@
+package api_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/mainflux/mainflux/manager"
+	"github.com/mainflux/mainflux/manager/api"
+	"github.com/mainflux/mainflux/manager/mocks"
+	"github.com/stretchr/testify/assert"
+)
+
+const contentType = "application/json; charset=utf-8"
+
+var (
+	validUser       = manager.User{"user@example.com", "password"}
+	invalidPassUser = manager.User{"user@example.com", "invalid_password"}
+	invalidUser     = manager.User{"userexample.com", "password"}
```

Comment:
This "fellow" exists for the sole purpose of providing a false e-mail value. Define a constant that contains "bad" value instead (this can be reused for other purposes as well, see other tests as an example).

The only exception to the aforementioned statement is user registration. For that purpose you can create local instance based on the `validUser` but using the constant I mentioned as an e-mail.